test/functional/ui/incsubstitution_spec.lua

```diff
@@ -0,0 +1,1374 @@
+    BufEnter = { open = {1, 1} },
+    BufFilePost = {},
+    BufFilePre = {},
+    BufHidden = { open = {2, 3} },
+    BufLeave = { open = {2, 3} },
+    BufNew = {},
+    BufNewFile = {},
+    BufRead = {},
+    BufReadCmd = {},
+    BufReadPre = {},
+    BufUnload = { open = {2, 3} },
+    BufWinEnter = {},
+    BufWinLeave = { open = {2, 3} },
+    BufWipeout = { open = {2, 3} },
+    BufWrite = {},
+    BufWriteCmd = {},
+    BufWritePost = {},
+    Syntax = {},
+    FileType = {},
+    WinEnter = { open = {1, 1} },
+    WinLeave = { open = {2, 3} },
+    CmdwinEnter = {},
+    CmdwinLeave = { open = {1, 1} },
+  }
+
+  local function bufferlist(t)
+    local s = ""
+    for _, buffer in pairs(t) do
+      s = s .. ", " .. tostring(buffer)
+    end
+    return s
+  end
+
+  -- fill the table with default values
+  for event, _ in pairs(eventsExpected) do
+    eventsExpected[event].open = eventsExpected[event].open or {}
+    eventsExpected[event].close = eventsExpected[event].close or {}
+  end
+
+
+  local function register_autocmd(event)
+    execute("let g:" .. event .. "_fired=[]")
+    execute("autocmd " .. event .. " * call add(g:" .. event .. "_fired, expand('<abuf>'))")
+  end
+
+  it('are run properly when splitting', function()
+    common_setup(nil, "split", default_text)
+
+    local eventsObserved = {}
+    for event, _ in pairs(eventsExpected) do
+      eventsObserved[event] = {}
+      register_autocmd(event)
+    end
+
+    feed(":%s/tw")
+
+    for event, _ in pairs(eventsExpected) do
+      eventsObserved[event].open = eval("g:" .. event .. "_fired")
+      execute("let g:" .. event .. "_fired=[]")
+    end
+
+    feed("/<enter>")
+
+    for event, _ in pairs(eventsExpected) do
+        eventsObserved[event].close = eval("g:" .. event .. "_fired") 
+    end
+
+    for event, _ in pairs(eventsExpected) do
+      eq(event .. " " .. bufferlist(eventsExpected[event].open),
+         event .. " " .. bufferlist(eventsObserved[event].open))
+      eq(event .. " " .. bufferlist(eventsExpected[event].close),
+         event .. " " .. bufferlist(eventsObserved[event].close))
+    end
+
+  end)
+
+  it('that close the incsub window', function()
+    local openEvents = {}
+
+    for event, _ in pairs(eventsExpected) do
+      if #eventsExpected[event].open > 0 then
+        openEvents[event] = true
+      end
+    end
+
+    local errorEvents = {
+      BufWipeout = "444",
+      BufDelete = "444",
+      BufUnload = "444",
+      BufWinLeave = "444",
+    }
+
+
+    local screen = Screen.new(50,10)
+
+    for event,_ in pairs(openEvents) do
+      clear()
+      common_setup(screen, "split", default_text)
+
+      execute("autocmd " .. event .. " * close!")
+      feed(":%s/.")
+
+      if errorEvents[event] then
+        feed("<enter>")
+        local err = errorEvents[event]
+        local errstr = string.find(eval('v:errmsg'), '^E' .. err .. ':')
+        neq(event .. "nil", event .. tostring(errstr))
```

Comment:
Why `neq`? It looks like it should be `eq(event .. err .. ':', event .. meths.get_vvar('errmsg'):sub(1, #err + 1))`: without `errstr` or any regexes involved.